sympy/utilities/codegen.py

```diff
@@ -1833,10 +1833,13 @@ def get_code_generator(language, project, standard=None):
             language = 'C89'
         elif standard.lower() == 'c99':
             language = 'C99'
-    CodeGenClass = {"C": CCodeGen, "C89": C89CodeGen, "C99": C99CodeGen,
-                    "F95": FCodeGen, "JULIA": JuliaCodeGen,
-                    "OCTAVE": OctaveCodeGen,
-                    "RUST": RustCodeGen}.get(language.upper())
+    if isinstance(language, type):
+        CodeGenClass = language
```

Comment:
I think we'll have to live with this for now. Seems like the CCodeGen class would need to be reworked in the long run to accept a printer instance in its `__init__` method for example.